transport/src/main/java/io/camunda/zeebe/transport/stream/impl/RemoteStreamPusher.java

```diff
@@ -58,7 +61,8 @@ private RemoteStreamErrorHandler<P> instrumentingErrorHandler(
     return (error, payload) -> {
       if (error != null) {
         metrics.pushFailed();
-        LOG.debug("Failed to push (size = {}) to stream {}", payload.getLength(), streamId, error);
+        pushErrorLogger.warn(
+            "Failed to push (size = {}) to stream {}", payload.getLength(), streamId, error);
```

Comment:
❓ Should the log level maybe different based on the error?

Reply:
Good question, I hadn't really thought about it. Let's walk through it.

- `NoSuchMemberException` -> thrown when we tried pushing to a gateway which was since removed from the cluster (or we have a bug, ofc :smile:). This is likely a temporary error - if the member is gone, its streams will get removed, but that's asynchronous so this may happen. It makes sense as a warning, since if it persists this is indicative of a bigger issue.
- `RemoteHandlerFailure` -> this is caused by an uncaught exception on the target side. This is typically pretty bad, but the relevant log will be on the receiver/target side. I think here also makes sense to log as a warning - the real error should be logged, as error, on the receiver side.
- `NoRemoteHandler` -> this is caused if the node isn't listening for pushes. This is either a bug (in which case warning makes sense), or temporary on shutdown, so it would resolve itself.
- `ProtocolException` -> this happens if the messaging service can't parse the request. Shouldn't happen, and if it does, it's likely a bigger issue with the complete messaging service, and we have other logs for that. We _could_ downgrade it to debug, but I think it's also OK as warning.

So as far as exceptions go, I think all of them as warning is OK.

Then we get `StreamResponseException`: basically a stream protocol error. There it will be based on the error code:

- `INTERNAL_ERROR` -> makes sense as warning, though likely won't recover, but the real error log should be on the receiver side.
- `NOT_FOUND` -> could be temporary, i.e. the stream was just closed asynchronously.
- `INVALID` -> means the request is invalid. For pushes this is not a possible case (yet anyway).
- `MALFORMED` -> could happen, means the push request we sent is not parse-able. I suppose it's unrecoverable, so here it probably makes sense as error level. wdyt?
- `EXHAUSTED` -> means all streams were tried unsuccessfully on the gateway (e.g. either they were closed as we tried pushing, or they were blocked, etc.). This could be debug since in most cases you won't care, but it could indicate that something really is wrong if this is returned all the time incorrectly :thinking: I'm not sure about this case.
- `BLOCKED` -> not yet in use, since even if the stream was blocked, we always return `EXHAUSTED` instead at the moment. Might be different later once we track and propagate capacity.

So what I see is that `MALFORMED` and `INVALID` (though not used here yet) make sense as errors, since the error is on the sender side. I don't know about `EXHAUSTED` yet :thinking:

Reply:
Also follow up decision, the throttled logger throttles regardless of log level. Should we use a different one, or refactor it so we throttle based on level? Since we probably want to show ERROR level more than WARN level errors.